### DISCUSSION TITLE
bullet 2.82 invalid normal direction on collision/sweep test

### DISCUSSION THREAD

**Discussion by shelomentsev** [bulletphysics/bullet3#3463] [General]
Sweep test vs btBvhTriangleMeshShape on some triangles returns normal in opositional direction.
I start digging this issue, I also found that collision boxes fall through that triangles(invalid contact normal).
If I change position one of triangle vertex then sweep return valid normal(same as box contact normal)
Here is source code tho show that issues(modified BasicDemo)
https://github.com/shelomentsev/bullet2_bug

**Comment by antarespilot** [comment]
I've having this problem as well. Boxes falls through about half of my triangles. 

It works using 2.81, but not with 2.82. This is a rather serious regression I would say.

**Comment by erwincoumans** [comment]
It should be fixed in latest github revision, did you try it?

**Comment by antarespilot** [comment]
Hi, just did, and everything works fine again. :-) Tested using double-precision-mode. 

I had not tried compiling from this repo in a while (using Linux), great to see all the things that have happened here during the last month or so.

**Comment by shelomentsev** [comment]
I just did too, bug not fixed.
If I set m_fixContactNormalDirection = 0, it fix bug
